### DISCUSSION TITLE
How to set last replaced for a past date ?

### DISCUSSION THREAD

**Discussion by Alfred99-HA** [andrew-codechimp/HA-Battery-Notes#471] [Q&A] [answered]
Actually I am looking for a method on how to set last replaced for a past date. For devices I know when I integrated and so far have not replaced the battery I would like to do so.
My idea was to set the state "last replaced" in the developer section properly, but after restart a new date shows up.
That's how I did set the dates
![replaced set manually](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/133593172/1c1da454-35af-451a-9ea4-ddf07c3d57b2)
After restart
![after restart](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/133593172/fe78c478-4f9d-489d-8a11-76409923825f)
![Clipboard_01-12-2024_01](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/133593172/61a46e90-e31b-4612-8267-622cd4e5c7ec)
Help how to accomplish is very much appreciated

**Comment by andrew-codechimp** [comment, marked as answer]
As long as you are running version 1.4 you can now do this via the Set battery replaced service.  There's now an optional date.

![CleanShot 2024-01-12 at 13 44 04@2x](https://github.com/andrew-codechimp/HA-Battery-Notes/assets/1849731/231078ac-7f5c-4fdd-b2b7-eb929abeba99)

I can't do it via the GUI as HA doesn't have a native date entry to change a state, I suppose it's quite a rare requirement.

That Set State in developer tools is very misleading, it only updates the state in memory for anything you set with it, and is truly just a dev tool for modifying state on the fly to detect changes. I only learnt this myself last week!

**Reply by Alfred99-HA** [reply]
Thanks a lot for immediate supporting of optional user defined date for past "Date replaced".
Furthermore thanks a lot for the hint reagarding dev tools!
